pkg/commands/ngwaf/workspace/threshold/update.go

```diff
@@ -0,0 +1,146 @@
+package threshold
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/fastly/go-fastly/v12/fastly"
+	"github.com/fastly/go-fastly/v12/fastly/ngwaf/v1/workspaces/thresholds"
+
+	"github.com/fastly/cli/pkg/argparser"
+	fsterr "github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/global"
+	"github.com/fastly/cli/pkg/text"
+)
+
+// UpdateCommand calls the Fastly API to update a workspace threshold.
+type UpdateCommand struct {
+	argparser.Base
+	argparser.JSONOutput
+
+	// Required.
+	thresholdID string
+	workspaceID argparser.OptionalWorkspaceID
+
+	// Optional.
+	action     argparser.OptionalString
+	dontNotify argparser.OptionalString
+	duration   argparser.OptionalInt
+	enabled    argparser.OptionalString
+	interval   argparser.OptionalInt
+	limit      argparser.OptionalInt
+	name       argparser.OptionalString
+	signal     argparser.OptionalString
+}
+
+// NewUpdateCommand returns a usable command registered under the parent.
+func NewUpdateCommand(parent argparser.Registerer, g *global.Data) *UpdateCommand {
+	c := UpdateCommand{
+		Base: argparser.Base{
+			Globals: g,
+		},
+	}
+	c.CmdClause = parent.Command("update", "Update a workspace threshold")
+
+	// Required.
+	c.RegisterFlag(argparser.StringFlagOpts{
+		Name:        argparser.FlagNGWAFWorkspaceID,
+		Description: argparser.FlagNGWAFWorkspaceIDDesc,
+		Dst:         &c.workspaceID.Value,
+		Action:      c.workspaceID.Set,
+	})
+	c.CmdClause.Flag("threshold-id", "Threshold ID").Required().StringVar(&c.thresholdID)
+
+	// Optional.
+	c.CmdClause.Flag("action", "The action to take when the threshold is exceeded. [block, log]").Action(c.action.Set).StringVar(&c.action.Value)
```

Comment:
For a future (large) PR, we should find an elegant way to avoid duplicating all of these flags in the `create` and `update` functions. This will require some way to indicate which ones are required and which are not, so it will take some design discussion, but the duplication causes a significant risk of mistakes in future changes.